### PR TITLE
Enable deploying templates to RoslynDev hive during build

### PR DESCRIPTION
**Pull request by mavasani**
Apart from making the core change to set `DeployExtension = true`, I had to make couple of additional changes:

1. Fix the AssemblyVersion, BuildVersion and VSIXVersion computation in the target. We ended up defaulting these to 1.0.0.0000101, which prevented locally built VSIXes from installing into the Dev hive due to it being lower then existing version. Now we default the versions to 42.42.42.42 when doing non-official builds.

2. Move to RC.2 VS SDK build tools - The preview4 build tools were crashing intermittently when deploying the VSIX during build.

### CODE REVIEW

**Comment by mavasani** [comment]
Tagging @dotnet/project-system for review

**Comment by mavasani** [comment]
Also tagging @dsplaisted

File: Common.props [hunk 1]
```diff
@@ -23,23 +23,72 @@
...
+    <!-- Split the build parts out from the BuildNumber which is given to us by MicroBuild in the format of yyyymmdd.nn
+         where BuildNumberFiveDigitDateStamp is mmmdd (such as 60615) and BuildNumberBuildOfTheDay is nn (which represents the nth build
+         started that day). So the first build of the day, 20160615.1, will produce something similar to BuildNumberFiveDigitDateStamp: 60615,
+         BuildNumberBuildOfTheDayPadded: 01;and the 12th build of the day, 20160615.12, will produce BuildNumberFiveDigitDateStamp: 60615, BuildNumberBuildOfTheDay: 12
+
+         Additionally, in order ensure the value fits in the 16-bit PE header fields, we will only take the last five digits of the BuildNumber, so
+         in the case of 20160615, we will set BuildNumberFiveDigitDateStamp to 60615. Unfortunately for releases in 2017 we can't go any higher, so
+         we will continue the month counting instead: the build after 61231 is 61301. -->
+    <BuildNumberFiveDigitDateStamp Condition="'$(BuildNumber)' != ''">$([MSBuild]::Subtract($(BuildNumber.Split('.')[0].Substring(3).Trim()), 8800))</BuildNumberFiveDigitDateStamp>
```
Comment (dsplaisted) [inline Common.props]: Is this subtracting `8800` from the build number, or am I misunderstanding?  What does the subtraction do?

Comment (dsplaisted) [inline Common.props]: > Unfortunately for releases in 2017 we can't go any higher, so we will continue the month counting instead: the build after 61231 is 61301.

Is this supposed to be handled by the logic here, or will MicroBuild pass us something like 20161301.1 as the BuildNumber?

**Review by dsplaisted — COMMENTED** [review]
I couldn't follow how the versioning logic went.  Beyond that this looks good.

File: Common.props (hunk 1 above)
Comment (mavasani) [inline Common.props]: This is basically replicating the logic that we used for Roslyn and Roslyn-project-system: https://github.com/dotnet/roslyn/blob/master/build/Targets/Versions.props#L43 @jasonmalinowski did the math magic here :-)

**Comment by mavasani** [comment]
FYI: I am merging this in so I can unblock John and Peter in their local VSI tests setup - attempting to install the VSIXes manually using VSIXExpInstaller seems to be flaky and causing unnecessary effort for them. I'll address any additional PR feedback with a follow up PR.

@srivatsn These are infrastructure changes, so I am assuming it doesn't need M2 approval.

**Pull request merged**